conan/tools/cmake/presets.py

```diff
@@ -158,6 +158,13 @@ def _configure_preset(conanfile, generator, cache_variables, toolchain_file, mul
                     "strategy": "external"
                 }
 
+        # Set the compiler like in the toolchain. Some IDEs like VS or VSCode require the compiler
+        # being set to cl.exe in order to activate the environment using vcvarsall.bat according to
```

Comment:
interesting! I've not found this documented for VS Code, but have verified that it indeed works that way. Thanks so much for figuring this out! :D

Reply:
I thought it was documented somewhere, but I couldn't find it either and can't remember how I learned about it in the first place. But I have used this trick successfully for some time now and seeing [this comment](https://github.com/microsoft/vscode-cmake-tools/issues/1929#issuecomment-858800902) this has been a thing for at least 3 years.